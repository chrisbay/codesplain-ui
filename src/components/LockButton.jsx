import React, { PropTypes } from 'react';

import IconButton from 'material-ui/IconButton';
import FontIcon from 'material-ui/FontIcon';
import LockOpen from 'material-ui/svg-icons/action/lock-open';
import Lock from 'material-ui/svg-icons/action/lock';

const LockButton = ({onClick, readOnly}) => {
  const lockIcon = readOnly ? <Lock /> : <LockOpen />;
  const toolTipText = readOnly ? "Switching back to edit mode not supported yet" : "Click to switch to prevent any changes to the snippet";
  return (
    <IconButton
      disabled={readOnly}
      label={`Switch to ${ readOnly ? 'edit' : 'read-only' } mode`}
      onTouchTap={onClick}
      tooltip={toolTipText}
    >
      {lockIcon}
    </IconButton>
  );
};

LockButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  readOnly: PropTypes.bool.isRequired,
};

export default LockButton;
