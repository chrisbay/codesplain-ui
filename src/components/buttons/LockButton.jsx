import React, { PropTypes } from 'react';

import IconButton from 'material-ui/IconButton';
import Lock from 'material-ui/svg-icons/action/lock';
import LockOpen from 'material-ui/svg-icons/action/lock-open';
import ReactTooltip from 'react-tooltip';

const styles = {
  button: {
    zIndex: 5,
  },
  buttonContainer: {
    height: '48px',
  },
};

const LockButton = ({ onClick, readOnly }) => {
  const lockIcon = readOnly ? <Lock /> : <LockOpen />;
  const toolTipText = readOnly ? 'Switching back to edit mode not supported yet' : 'Click to lock the snippet (this will prevent further changes.)';
  return (
    <span style={styles.buttonContainer}>
      <IconButton
        id="lockButton"
        disabled={readOnly}
        data-tip
        data-for="lock-tip"
        label={`Switch to ${readOnly ? 'edit' : 'read-only'} mode`}
        onTouchTap={onClick}
        style={styles.button}
      >
        {lockIcon}
      </IconButton>
      <ReactTooltip id="lock-tip" effect="solid">
        {toolTipText}
      </ReactTooltip>
    </span>
  );
};

LockButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  readOnly: PropTypes.bool.isRequired,
};

export default LockButton;
