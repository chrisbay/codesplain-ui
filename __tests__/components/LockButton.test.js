import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';

import LockButton from '../../src/components/LockButton';

describe('LockButton />', () => {
  it('matches snapshot of when the app is not in read-only mode', () => {
    const renderer = ReactTestUtils.createRenderer();
    const tree = renderer.render(
      <LockButton
        onClick={jest.fn()}
        readOnly={false}
      />
    );
    expect(tree).toMatchSnapshot();
  });
  it('matches snapshot of when the app is in read-only mode', () => {
    const renderer = ReactTestUtils.createRenderer();
    const tree = renderer.render(
      <LockButton
        onClick={jest.fn()}
        readOnly={true}
      />
    );
    expect(tree).toMatchSnapshot();
  });
})
