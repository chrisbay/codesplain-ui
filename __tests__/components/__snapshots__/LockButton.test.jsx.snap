exports[`<LockButton /> matches snapshot of when the app is in read-only mode 1`] = `
<div
  title="Switching back to edit mode not supported yet">
  <IconButton
    disableTouchRipple={false}
    disabled={true}
    iconStyle={Object {}}
    label="Switch to edit mode"
    style={
      Object {
        "zIndex": 5,
      }
    }
    tooltipPosition="bottom-center"
    touch={false}>
    <ActionLock />
  </IconButton>
</div>
`;

exports[`<LockButton /> matches snapshot of when the app is not in read-only mode 1`] = `
<div
  title="Click to lock the snippet (this will prevent further changes.)">
  <IconButton
    disableTouchRipple={false}
    disabled={false}
    iconStyle={Object {}}
    label="Switch to read-only mode"
    style={
      Object {
        "zIndex": 5,
      }
    }
    tooltipPosition="bottom-center"
    touch={false}>
    <ActionLockOpen />
  </IconButton>
</div>
`;
